src/ray/common/buffer.h

```diff
@@ -30,9 +33,9 @@ class Buffer {
 /// Represents a byte buffer in local memory.
 class LocalMemoryBuffer : public Buffer {
  public:
-  LocalMemoryBuffer(uint8_t *data, size_t size, bool should_copy = false)
-      : data_(data), size_(size) {
-    if (should_copy) {
+  LocalMemoryBuffer(uint8_t *data, size_t size, bool has_data_copy = false)
```

Comment:
I don't really understand why this new flag is needed from the code. Can you please document what exactly the semantics are for this new flag and also why it's needed? Depending on exactly how it's used, it might be better to add another `Buffer` implementation instead of this flag - I'm generally in favor of minimizing the use of boolean flags.

Reply:
this is not really a new flag. it's just a rename of existing flag. Originally we have a different type of buffer to cover the case where the buffer owns the data, but we found it's hard to find appropriate names to differentiate the two types, so we ended up with adding a flag. Kindly let me know if you have suggestions on the naming. Thanks.